portfolio/src/main/webapp/script.js

```diff
@@ -79,17 +79,29 @@ async function getData(path) {
   // When the request is complete return text from response
   return data;
 }
-async function includeHTML(fileName)
+async function include(path,extension)
 {
-    const element = document.getElementById(`include-${fileName}`);
+    const element = document.getElementById(`include-${path}`);
     if(element)
     {
-        const header = await getData(`/${fileName}.html`);
+        const header = await getData(`/${path}${extension}`);
         element.innerHTML = header+'\n'+element.innerHTML;
     }
 }
+async function includeHTML(fileName)
+{
+    include(fileName,".html");
+}
+async function includeDynamicHTML(fileName)
+{
+    include(fileName,"");
+}
 
 document.addEventListener('DOMContentLoaded',() => {
     includeHTML('navigation');
     includeHTML('contact');
+    includeDynamicHTML('data');
+    getData('/data').then((data) => {
+        console.log(data);
+    });
```

Comment:
Nit: Try not to leave `console.log` statements in your finished code. It's great for debugging now but too many log statements will get really hard to read through on bigger projects.

Seeing as you're calling `getData('/data')` just to log the output, I'd recommend deleting lines 104 - 106. Feel free to add `getData('/data')` in a future PR when you're ready to use the output of that call.

Reply:
yea, I normally wouldn't have left that there, I only did because it was one of the instructions in the tutorial so I wanted to leave it there to show I actually did that step

Reply:
will remove in next pr